sample/Evernote Business/server.py

```diff
@@ -0,0 +1,233 @@
+app.config['SECRET_KEY'] = "secret key" #configure the secret key for session data (must be the same secret key when starting Flask in the last line)
+
+@app.route("/auth")
+def auth():
+	"""Takes a callback for Evernote's OAuth process 
+
+	Redirect to main after login is confirmed an token is stored in the session"""
+	
+	#check to make sure the user approved the appliction (oauth_verifier will not be present if they declined)
+	if "oauth_verifier" in request.args:
+		#setup client
+		client = EvernoteClient(
+		consumer_key=CONSUMER_KEY,
+		consumer_secret=CONSUMER_SECRET,
+		sandbox= sandbox
+		)
+		
+		#get access token
+		try:
+			auth_token = client.get_access_token(session['oauth_token'], session['oauth_token_secret'], request.args['oauth_verifier'])
+		except:
+			return render_template("error.html", error_message="OAuth Error: Please approve access to the appliction %s" % CONSUMER_KEY)	
+		
+		#attach the user's access token to the session
+		session["access_token"]=auth_token
+
+		#redirect to main
+		return redirect(url_for("main"))
+	
+	#If the user did not approve access to our application let the user know
+	else:
+		return render_template("error.html", error_message="OAuth Error: Please approve access to the appliction %s" % CONSUMER_KEY)
+
+
+@app.route("/", methods=['POST','GET'])
+def main():
+	"""Takes the access token in the session and presents the user with notes that are tagged "template" or "Template" 
+
+	If there are no notes tagged "template" or "Template" 4 example notes are created
+
+	If the user is not logged in: a splash page with the option to authorize the appliction is presented
+	"""
+
+	#check to see the user has logged in
+	if "access_token" in session.keys():
+		
+		try:
+			#setup Evernote client
+			client = EvernoteClient(token=session["access_token"], sandbox=sandbox)
+			user_store = client.get_user_store()
+			note_store = client.get_note_store()			
+		except Errors.EDAMSystemException:
+			#if the user authentication token fails to work prompt them to reautenticate
+			return render_template("error.html", error_message="Your session is no longer valid.  Please click here <a href=\'/clear'>click here</a> to reset and login again.")
+
+		#if the user is not a part of the business, tell the user
+		user = user_store.getUser()
+		if not user.accounting.businessId:
+			return render_template("error.html", error_message="This account is not a part of a business.  Please click here <a href=\'/clear'>click here</a> to reset and try another account.")
+		else:
+			business_store = client.get_business_note_store()
+
+		#setup search
+		notebook_filter=NoteStoreTypes.NoteFilter()
+		result_spec = NoteStoreTypes.NotesMetadataResultSpec(includeTitle=True)
+
+		# Start seraching for notes in the last day
+		# if less than 5 results increase to 10, 100, etc. until more than 5 are found or 
+		# the number results between iterations are the same
+		search_results_len = None
+		while search_results_len<10:
```

Comment:
You don't need to re-run the search results with a larger set until you find something - if the first search returns only 3 results then every search thereafter will return the same 3 notes. Additionally, the first search will return a count of all notes that matched the provided query in the [`totalNotes` field of the `NotesMetadataList`](https://dev.evernote.com/doc/reference/NoteStore.html#Struct_NotesMetadataList) returned by `findNotesMetadata()`